server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1439,22 +1440,53 @@ public ReplicationCheckpoint getLatestReplicationCheckpoint() {
     }
 
     /**
-     * Checks if checkpoint should be processed
-     *
-     * @param requestCheckpoint       received checkpoint that is checked for processing
-     * @return true if checkpoint should be processed
+     * Checks if this shard is able to perform segment replication.
```

Comment:
nit:
> * Checks if this shard is able to perform segment replication.

Checks if this target shard should start round of segment replication with primary ?

Reply:
Updated - I'm intentionally not mentioning primary here as these paths will be reused for remote store replication.